Source/WebCore/animation/AnimationEffect.cpp

```diff
@@ -73,7 +73,7 @@ EffectTiming AnimationEffect::getBindingsTiming() const
     else
         timing.duration = autoAtom();
     timing.direction = m_timing.direction;
-    timing.easing = m_timing.timingFunction->cssText();
+    timing.easing = m_timing.timingFunction.get()->cssText();
```

Comment:
`m_timing.timingFunction` is a `RefPtr<TimingFunction>`. Why is this not safe to call the `TimingFunction::cssText()` method on and how calling it on the raw pointer safer?

Reply:
Yeah, I don't think this is correct. We need to store it in a local stack variable which is RefPtr instead.
The problem with calling a non-trivial function like `cssText` is that it could alter memory state. Out of caution, we need to store it in RefPtr in stack while making a function call.
See https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines#when-calling-a-non-trivial-member-function-on-an-object-hold-a-smart-pointer-to-the-object-on-the-stack